api/bolt/migrator/migrate_dbversion23.go

```diff
@@ -7,10 +7,18 @@ func (m *Migrator) updateSettingsToDB24() error {
 	if err != nil {
 		return err
 	}
-
+	updateSettings := false
```

Comment:
No need to add check on update, the migration is lightweight enough and only runs once. 

Also I believe that the condition will never happen so no need for it here `if legacySettings.UserSessionTimeout == "" {`, it should be assigned for any migration.